mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -4408,6 +4408,32 @@ func TestAccContainerCluster_withEnablePrivateEndpointToggle(t *testing.T) {
 	})
 }
 
+func TestAccContainerCluster_failedCreation(t *testing.T) {
+	// Test that in a scenario where the cluster fails to create, a subsequent apply will delete the resource.
+	t.Parallel()
+
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+
+	project := BootstrapProject(t, "tf-test-project-fail-cluster", getTestBillingAccountFromEnv(t), []string{"container.googleapis.com"})
+	removeContainerServiceAgentRoleFromContainerEngineRobot(t, project)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccContainerCluster_failedCreation(clusterName, project.ProjectId),
+				ExpectError: regexp.MustCompile(".*timeout while waiting for state to become 'DONE'.*"),
```

Comment:
Do you need the `.*`s for these regexes? My understanding was that matching a subsection was valid, not necessarily the whole string.

Reply:
I have removed the `.*`s.